### PR TITLE
Adding support for aws request signing

### PR DESCRIPTION
**Pull request by Mzack9999**
## Proposed changes

This PR adds a prototype support for AWS requests signing 

## Checklist
- [x] Pull request is created against the [dev](https://github.com/projectdiscovery/nuclei/tree/dev) branch
- [x] All checks passed (lint, unit/integration/regression tests etc.) with my changes (some tests fail due to known issues)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by Mzack9999** [comment]
template example:
```
```

**Comment by Mzack9999** [comment]
Indirectly fixes #844

File: v2/pkg/protocols/http/request.go [hunk 1]
```diff
@@ -391,9 +391,48 @@ func (request *Request) executeRequest(reqURL string, generatedRequest *generate
 			}
 		}
 		if resp == nil {
+			switch request.Signature.Value {
```
Comment (Ice3man543) [inline v2/pkg/protocols/http/request.go:394]: we can move this code to a separate function

Comment (Mzack9999) [reply v2/pkg/protocols/http/request.go]: done

**Review by Ice3man543 — APPROVED** [review]
lgtm!

**Review by ehsandeep — CHANGES_REQUESTED** [review]
```console
go run . -duc -no-interactsh -t aws-test.yaml

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.5.4

		projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Using Nuclei Engine 2.5.4 (outdated)
[INF] Using Nuclei Templates 8.7.3 (latest)
[INF] Using Interactsh Server https://interactsh.com
[INF] Templates added in last update: 5
[INF] Templates loaded for scan: 1
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x20 pc=0x10325b010]

goroutine 55 [running]:
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*generatedRequest).URL(...)
	/Users/geekboy/Github/nuclei/v2/pkg/protocols/http/build_request.go:46
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*Request).ExecuteWithResults.func1({0x1400020e180, 0x162}, 0x0, 0x14000072150)
	/Users/geekboy/Github/nuclei/v2/pkg/protocols/http/request.go:253 +0x1c0
github.com/projectdiscovery/nuclei/v2/pkg/protocols/http.(*Request).ExecuteWithResults(0x140000e2420, {0x0, 0x0}, 0x14000072150, 0x140000721b0, 0x14000072330)
	/Users/geekboy/Github/nuclei/v2/pkg/protocols/http/request.go:315 +0x280
github.com/projectdiscovery/nuclei/v2/pkg/protocols/common/executer.(*Executer).Execute(0x140004bfd60, {0x0, 0x0})
	/Users/geekboy/Github/nuclei/v2/pkg/protocols/common/executer/executer.go:51 +0x1c4
github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).executeSelfContainedTemplateWithInput(0x140000e2160, 0x140004cf6c0, 0x1400003b400)
	/Users/geekboy/Github/nuclei/v2/pkg/core/execute.go:60 +0x38
github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).ExecuteWithOpts.func1(0x140000e2160, 0x1400003b400, 0x3, {0x103bf16d8, 0x140004be080}, 0x140004bed60, 0x140004cf6c0)
	/Users/geekboy/Github/nuclei/v2/pkg/core/execute.go:46 +0x40
created by github.com/projectdiscovery/nuclei/v2/pkg/core.(*Engine).ExecuteWithOpts
	/Users/geekboy/Github/nuclei/v2/pkg/core/execute.go:42 +0x13c
exit status 2
```

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**